src/Sarif.Viewer.VisualStudio/ErrorListService.cs

```diff
@@ -0,0 +1,155 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+using Microsoft.CodeAnalysis.Sarif.Driver;
+using Microsoft.CodeAnalysis.Sarif.Driver.Sdk;
+using Microsoft.CodeAnalysis.Sarif.Sdk;
+
+namespace SarifViewer
+{
+    public class ErrorListService
+    {
+        public static ErrorListService Instance = new ErrorListService();
+
+        public static void ProcessSarifLog(ResultLog resultLog)
+        {
+            foreach (RunLog runLog in resultLog.RunLogs)
+            {
+                Instance.WriteRunLogToErrorList(runLog);
+            }
+
+            SarifTableDataSource.Instance.BringToFront();
+        }
+
+        private ErrorListService()
+        {
+            this.documentToLineIndexMap = new Dictionary<string, NewLineIndex>();
+        }
+
+        private Dictionary<string, NewLineIndex> documentToLineIndexMap;
+
+        private IRuleDescriptor GetRule(RunLog runLog, string ruleId)
+        {
+            if (runLog.RuleInfo == null)
+            {
+                return null;
+            }
+
+            foreach (IRuleDescriptor ruleDescriptor in runLog.RuleInfo)
+            {
+                if (ruleDescriptor.Id == ruleId) { return ruleDescriptor; }
+            }
+
+            throw new InvalidOperationException();
```

Comment:
You started the validation logic a while back and then we abandoned it because we needed to recover Bill's SDK work. I'd be very interested to get that going again. :) It belongs somewhere in sarif-sdk.